### PR TITLE
Implement a multi-select component

### PR DESCRIPTION
**Pull request by carlosepp**
- Refactors react-select suggested stylesheet to use custom variables for customization
- The ability to select more than one item from a dropdown list
- The necessary classes to highlight multiple items at the same time

### CODE REVIEW

File: components/_form.scss [hunk 1]
```diff
@@ -154,6 +167,262 @@ $include-html-paint-form: true !default;
...
+  border-radius: $global-radius;
+  color: $form-multi-select-item-color;
+  float: left;
+  font-size: $small-font-size;
+  margin: $column-gutter / 5 $column-gutter / 5 $column-gutter / 5 0;
+  padding: 0;
+  position: relative;
+
+  &:hover {
+    .item-label {
```
Comment (deioo) [inline components/_form.scss]: Since we're inside the item, we could rename this to `.label` and `.icon` below

File: components/_form.scss [hunk 2]
```diff
@@ -154,6 +167,262 @@ $include-html-paint-form: true !default;
...
+@mixin form-multi-select-options() {
+  background-color: $white;
+  border: 1px solid;
+  border-color: $form-input-border-color;
+  border-radius: 0 0 $global-radius $global-radius;
+  box-shadow: 0 1px 0 rgba(0, 0, 0, .06);
+  box-sizing: border-box;
+  margin-bottom: $column-gutter / 2;
+  margin-top: -1px;
+  max-height: $form-multi-select-menu-max-height;
```
Comment (deioo) [inline components/_form.scss]: `multi-select-options` instead of `menu`

File: components/_form.scss [hunk 3]
```diff
@@ -154,6 +167,262 @@ $include-html-paint-form: true !default;
...
+    position: relative;
+
+    &.is-open > .control {
+      border-bottom-left-radius: 0;
+      border-bottom-right-radius: 0;
+      border-color: $form-input-border-color-focused;
+    }
+
+    &:not(.is-open) > .options-selector {
+      display: none;
```
Comment (deioo) [inline components/_form.scss]: We should `display: none` by default and display:block when `.is-open`

File: components/_form.scss [hunk 4]
```diff
@@ -339,6 +608,10 @@ $include-html-paint-form: true !default;
       }
     }
 
+    &.form-multi-select {
```
Comment (deioo) [inline components/_form.scss]: Let's move this next to the `form-select` declaration for consistency

File: components/_form.scss (hunk 1 above)
Comment (carlosepp) [inline components/_form.scss]: Thought about that too :ok_hand:

File: components/_form.scss [hunk 5]
```diff
@@ -154,6 +167,298 @@ $include-html-paint-form: true !default;
...
+  .noresults {
+    box-sizing: border-box;
+    color: $iron;
+    cursor: default;
+    display: block;
+    line-height: $form-input-height;
+    padding-right: $column-gutter / 3;
+  }
+
+  // refactor and showcase
```
Comment (deioo) [inline components/_form.scss]: @carlosepp since we're not using this it should be ok to remove for now.

File: components/_form.scss [hunk 6]
```diff
@@ -154,6 +167,298 @@ $include-html-paint-form: true !default;
...
+    margin: $column-gutter / 4  0;
+    right: 0;
+
+    &:hover {
+      color: $info-color;
+    }
+  }
+
+  #{$options-selector} {
+    @include form-multi-select-options($options: $options, $option: $option, $is-focused-selector: $is-focused-selector);
```
Comment (deioo) [inline components/_form.scss]: Let's move the params one per row

**Comment by deioo** [comment]
After the comments are addressed, feel free to add a changelog and release :+1:

File: components/_form.scss (hunk 5 above)
Comment (carlosepp) [inline components/_form.scss]: :ok_hand:

**Pull request merged**